Extending/Packages/Packages-on-Umbraco-Cloud/index.md

```diff
@@ -0,0 +1,271 @@
+
+So at this point I have my one local site, however to test this I will push this all to the Cloud site and then clone it down again.
+
+At this point I have two local sites, one cloned down with the uaas.cmd tool. Meaning it has a visual studio solution and a class library, and the other is a regular clone I will spin up in VS Code.
+
+Site 1:
+Full Visual Studio solution
+Running on http://localhost:6240/ (Randomly generated)
+Has the ValueConnector in a class library that is built to a dll and copied to the websites bin on build
+
+Site 2: 
+A website served through VS Code (Could be IIS or anything else, doesn't matter)
+Running on http://localhost:17025/ (Randomly generated)
+Has the ValueConenctor dll in the bin from the clone
+
+Now we will set up these two identical sites to transfer content to eachother, to do so go to `site1/Config/UmbracoDeploy.config` and edit the live environment url to be Site2's url (http://localhost:17025/ in my case).
+Then do the same for site2 but put in site1's domain as the "live" one.
+
+At this point you should be able to go to the backoffice of either environment and do a Content transfer to live, and it should end up on the other (Assuming no errors from your custom connector).
+
+## Debugging a transfer
+
+At this point we haven't done anything to the ValueConverter yet, other than return the original value. Now we will attach Visual Studio to the IIS processes and try a transfer to see what it sends along.
+
+Go to Visual Studio, hit "Attach to Process" (default ALT + CTRL + P), and choose your two IIS processes. Add breakpoints in the `ToArtifact` and `FromArtifact` methods and go to site2, then try to do a content transfer to live (site2).
+
+It will hit your breakpoint, and if you continue you will then get an error. On the breakpoint you can see why though, it should look like this:
+
+![Hitting the breakpoint](images/hitting-breakpoints.png)
+
+Here you can see that value is null, so if you try to return `value.ToString()` you will get a null exception. So we will change the `ToArtifact` method a little:
+
+```csharp
+public string ToArtifact(object value, PropertyType propertyType, ICollection<ArtifactDependency> dependencies)
+{
+    var svalue = value as string;
+    if (string.IsNullOrWhiteSpace(svalue))
+        return null;
+
+    return value.ToString();
+}
+```
+
+But at this point you can't reattach the process as the code has changed. So we build, go to site1/bin and copy projectalias.Core.dll and projectalias.Core.pdb into site2/bin and then attach to process and try another transfer.
+
+The workflow here is not optimal, but a lot quicker than trying to deploy to Cloud everytime, and with this you can attach the debugger as well to help you out.
+
+Let's sync up, attach the debugger and attempt another transfer. Now you will see that `value` is null a few times, then your hardcoded id a few times, but nothing breaks here. Suddenly you will hit the `FromArtifact` method instead:
```

Comment:
What exactly does "Let's sync up" mean here? 
Does it mean that everything needs to be pushed to the Cloud environment, and then pulled down again, so the sites are in sync? 🤔 

```suggestion
Let's sync up, attach the debugger and attempt another transfer. Now you will see that `value` is null a few times, then your hardcoded ID a few times, but nothing breaks here. Suddenly you will hit the `FromArtifact` method instead:
```